packages/scaffold-config/src/frameworks.ts

```diff
@@ -14,10 +14,42 @@ export type WizardBundler = typeof dependencies.WIZARD_BUNDLERS[number]
 
 export type CodeGenFramework = Cypress.ResolvedComponentFrameworkDefinition['codeGenFramework']
 
+let yarnPnpRegistrationPath: string | undefined
+
+async function readPackageJson (packageFilePath: string): Promise<PkgJson> {
+  if (yarnPnpRegistrationPath) {
+    // using Yarn PnP. You cannot `fs.readJson`, since the module is zipped.
+    // use require.resolve - The PnP runtime (.pnp.cjs) automatically patches Node's
+    // fs module to add support for accessing files inside Zip archives.
+    // @see https://yarnpkg.com/features/pnp#packages-are-stored-inside-zip-archives-how-can-i-access-their-files
+    return require(require.resolve(packageFilePath))
+  }
+
+  return await fs.readJson(packageFilePath)
+}
+
 export async function isDependencyInstalled (dependency: Cypress.CypressComponentDependency, projectPath: string): Promise<Cypress.DependencyToInstall> {
   try {
     debug('detecting %s in %s', dependency.package, projectPath)
 
+    // we only need to register this once, when the project check dependencies for the first time.
+    if (yarnPnpRegistrationPath !== projectPath) {
+      try {
+        const pnpapi = require(path.join(projectPath, '.pnp.cjs'))
```

Comment:
`.pnp.cjs` is regenerated by Yarn every time you install a module. We only want to require it once, since the trick Yarn PnP uses is to dynamically monkeypatch `module` and do some other weird hacks to facilitate the various features of PnP.

It's a bit of a hack but it does run really fast, although Yarn 1 -> Yarn 2.x + PnP is a heck of a lift for a lot of code bases and projects, which is why many people are stuck on v1 (including FB, the (original) authors of Yarn).